server/events/approve_policies_command_runner.go

```diff
@@ -63,6 +66,16 @@ func (a *ApprovePoliciesCommandRunner) Run(ctx *command.Context, cmd *command.Co
 
 	result := a.buildApprovePolicyCommandResults(ctx, projectCmds)
 
+	// Adds the policy check output for failing policies which needs to be populated when using github checks
+	// Noop when github checks is not enabled.
+	policyCheckOutputStore := a.policyCheckOutputGenerator.GenerateCommandOutput(ctx, cmd)
+	for i, prjResult := range result.ProjectResults {
+		policyCheckOutput := policyCheckOutputStore.GetOutputFor(prjResult.ProjectName, prjResult.Workspace)
+		if policyCheckOutput != nil {
+			result.ProjectResults[i].PolicyCheckSuccess = policyCheckOutputStore.GetOutputFor(prjResult.ProjectName, prjResult.Workspace)
+		}
+	}
```

Comment:
The goal of adding this policy check output is to retain the information about the failing policies after they are appproved so yes, the output section will look like the policies are still failing. However, the description will have information to indicate that the policy check has passed. Here's a demo checkrun that shows failing policies after they have been approved: https://github.com/Altlantis-Dev/Test/pull/126/checks?check_run_id=7830917467

Regarding the templates, the only change we're making is changing the content of the policy check output from "Policies Approved" to the failing policies. And since our template resolved uses `PolicyCheckSuccess != nil` check to resolve template, I think we should be fine. The only issue I saw is the [metadata](https://github.com/lyft/atlantis/blob/2b99a2cb5cc16fd95b049d731b7fa11e86febcac/server/events/project_command_runner.go#L252-L257) being passed by PolicyCheck which is used in the template. So, I have added if conditionals to check before using that metadata in the template. 

So. to sum it up, the only noticeable difference in the checkrun output for policy_check and policy_approval is the bottom [part](https://github.com/lyft/atlantis/blob/7ee7fba93e34f32772765307f9e1537aec5b81fb/server/vcs/markdown/templates/policyCheckSuccessUnwrapped.tmpl#L5-L9).

Reply:
Since we don't do any computation in project level [approve_policies](https://github.com/lyft/atlantis/blob/2b99a2cb5cc16fd95b049d731b7fa11e86febcac/server/events/project_command_runner.go#L207-L215), we don't have to worry about policy approval failing for a project. The overall command can fail (when the approving user is not a policy owner for eg.) but the project level approve_policies should always pass. And since we use template resolver for project level checkruns, we don't have to worry about dealing with project level approve_policies command failure.